chainer/functions/noise/dropout.py

```diff
@@ -25,10 +25,18 @@ def forward(self, x):
             xp = cuda.get_array_module(*x)
             if xp == numpy:
                 flag = xp.random.rand(*x[0].shape) >= self.dropout_ratio
+                self.mask = scale * flag
             else:
-                flag = (xp.random.rand(*x[0].shape, dtype=numpy.float32) >=
-                        self.dropout_ratio)
-            self.mask = scale * flag
+                rand = xp.random.rand(*x[0].shape, dtype=numpy.float32)
+                self.mask, y = cuda.elementwise(
+                    'T x, R r, T scale, T ratio', 'T mask, T y',
+                    '''
+                    mask = (r >= ratio) * scale;
+                    y = x * mask;
+                    ''',
+                    'dropout_fwd',
+                )(x[0], rand, scale, self.dropout_ratio)
+                return y,
         return x[0] * self.mask,
```

Comment:
`if` constructs and `return` statements are not balanced. It's hard to read.
How about this?

(I know `x[0] * self.mask` is repeated. But IMO it's better than skewed code.)


```
    def forward(self, x):
        self.retain_inputs(())
        y = None
        if hasattr(self, 'mask'):
            y = x[0] * self.mask
        else:
            scale = x[0].dtype.type(1. / (1 - self.dropout_ratio))
            xp = cuda.get_array_module(*x)
            if xp == numpy:
                flag = xp.random.rand(*x[0].shape) >= self.dropout_ratio
                self.mask = scale * flag
                y = x[0] * self.mask
            else:
                rand = xp.random.rand(*x[0].shape, dtype=numpy.float32)
                self.mask, y = cuda.elementwise(
                    'T x, R r, T scale, T ratio', 'T mask, T y',
                    '''
                    mask = (r >= ratio) * scale;
                    y = x * mask;
                    ''',
                    'dropout_fwd',
                )(x[0], rand, scale, self.dropout_ratio)
        return y,
```